libctru/source/services/nim.c

```diff
@@ -0,0 +1,254 @@
+#include <string.h>
+#include <stdlib.h>
+#include <malloc.h>
+#include <3ds/types.h>
+#include <3ds/result.h>
+#include <3ds/svc.h>
+#include <3ds/srv.h>
+#include <3ds/synchronization.h>
+#include <3ds/services/nim.h>
+#include <3ds/ipc.h>
+#include <3ds/util/utf.h>
+
+static Handle nimsHandle;
+static int nimsRefCount;
+
+static Result NIMS_GetInitializeResult(void)
+{
+	Result ret = 0;
+	u32 *cmdbuf = getThreadCommandBuffer();
+
+	cmdbuf[0] = IPC_MakeHeader(0x3f, 0, 0); // 0x003f0000
+
+	if (R_FAILED(ret = svcSendSyncRequest(nimsHandle))) return ret;
+
+	return (Result)cmdbuf[1];
+}
+
+static Result NIMS_RegisterSelf(void)
+{
+	Result ret = 0;
+	u32 *cmdbuf = getThreadCommandBuffer();
+
+	cmdbuf[0] = IPC_MakeHeader(0x3c, 0, 2); // 0x003c0002
+	cmdbuf[1] = IPC_Desc_CurProcessHandle();
+	cmdbuf[2] = IPC_Desc_MoveHandles(1);
+
+	if (R_FAILED(ret = svcSendSyncRequest(nimsHandle))) return ret;
+
+	return (Result)cmdbuf[1];
+}
+
+static Result NIMS_SetTIN(void)
+{
+	Result ret = 0;
+	u32 *cmdbuf = getThreadCommandBuffer();
+	static char attr[] = "TIN";
```

Comment:
This is hardcoded because `nim:s` is only used by tiger, so only one TIN value is ever valid in this scenario. I just tested it and `nim` *does* accept other TINs, even if they're the wrong ones, so I can make this a parameter and have a user-friendly wrapper with this default instead.